test/llvmcall.jl

```diff
@@ -147,3 +147,34 @@ function confuse_declname_parsing()
     Void, Tuple{})
 end
 confuse_declname_parsing()
+
+
+module ObjLoadTest
+    using Base.Test
+    using Base.llvmcall
+    here = dirname(@__FILE__)
+    didcall = false
+    function callback()
+        global didcall
+        didcall = true
+        nothing
+    end
+    Base.ccallable(callback,Void,Tuple{},:jl_the_callback)
+    Base.ccallable(callback,Void,Tuple{},:_jl_the_callback)
+    # For now, only test this on Unix. On Windows we expect
+    # special structure in the object file to allow us to unwind,
+    # which this object file does not have.
+    @unix_only ccall(:jl_load_object_file,Void,(Ptr{UInt8},),joinpath(here,"objloadtest/objloadtest.o"))
+    function do_the_call()
+        llvmcall(
+        (""" declare void @call_the_callback_back()""",
+        """
+        call void @call_the_callback_back()
+        ret void
+        """),Void,Tuple{})
+    end
+    @unix_only begin
+	do_the_call()
+    	@test didcall
+    end
```

Comment:
it seems like ccallable could be tested easier without an object file:

```julia
julia> callback() = (global didcall = true; nothing)
julia> Base.ccallable(callback,Void,Tuple{},:the_ccallback)
julia> function do_the_call()
       llvmcall(
       (""" declare void @the_ccallback()""",
       """
       call void @the_ccallback()
       ret void
       """), Void,Tuple{})
       end
julia> do_the_call()
julia> @test didcall
```

for the object file, i think we could commit a `.ll` file. but i think this example makes me much less inclined to want to merge this since it is highly convoluted and the behavior seems to be a bit ill-defined and implementation dependent. it seems to be reimplementing shared libraries (we already have native interop tests for ccall dynamic libraries).